robotfm_tests/chatops_slack_check.rst

```diff
@@ -0,0 +1,107 @@
+        Should Contain     ${result}  ${SLACKCAT_TOKEN}
+
+    Configure ChatOps env with Slack token
+       Run Keyword      Add slack token at ${ST2CHATOPS.ENV}
+       ${result}=       Grep File    ${ST2CHATOPS.ENV}  export HUBOT_SLACK_TOKEN\=${HUBOT_SLACK_TOKEN}
+       Log To Console   \nTOKEN: ${result}
+       Should Contain   ${result}    export HUBOT_SLACK_TOKEN\=${HUBOT_SLACK_TOKEN}
+       ${result}=       Grep File    ${ST2CHATOPS.ENV}  export HUBOT_ADAPTER\=slack
+       Log To Console   \nADAPTER: ${result}
+       Should Contain   ${result}    export HUBOT_ADAPTER\=slack
+
+    Restart and check st2chatops service
+        ${result}=         Run Process    sudo  service  st2chatops  restart
+        Log To Console     \nSTDOUT: ${result.stdout} \nSTDERR: ${result.stderr} \nRC ${result.rc}
+        ${result}=         Run Process    service  st2chatops  status
+        Log To Console     \nSTDOUT: ${result.stdout} \nSTDERR: ${result.stderr} \nRC ${result.rc}
+        Should Contain     ${result.stdout}    running  
+
+    Check post_message execution and receive status
+        ${channel}=       Generate Token
+        # Log To Console   \nCHANNEL: ${channel}
+        ${result}=       Run Keyword  Hubot Post  ${channel}
+        # Log To Console   \nSTDOUT: ${result.stdout} \nSTDERR: ${result.stderr} \nRC ${result.rc}
+        Should Contain   ${result.stdout}   Chatops message received
+        Should Contain   ${result.stdout}   ${channel}
+ 
+    Execution success    
+        ${result}=         Run Keyword  Get last post_message execution id for slackcat
+        Sleep  10s
+        Should Contain     ${result.stdout}  slackcat posted 1 message lines to chatopsci
+        Log To Console     SLACKCAT: \nSTDOUT: ${result.stdout} \nSTDERR: ${result.stderr} \nRC ${result.rc}
+        #${result}=         Run Keyword  Get last post_message execution id for slackcat          
+
+
+
+
+    *** Keyword ***
+    Hubot Post
+        [Arguments]    ${channel}
+        ${result}=     Run Process    {  echo  -n;  sleep  5;  st2  action  execute  chatops.post_message  channel\=${channel}
+        ...                           message\='Debug. If you see this you are incredibly lucky but please ignore.'
+        ...                           >\/dev\/null;  echo;  sleep  2;}  |  bin\/hubot  \-\-test
+        ...                           cwd=/opt/stackstorm/chatops/    shell=True
+        [return]       ${result}
+
+    Generate Token
+        ${token}=      Generate Random String  32
+        [return]       ${token}
+    
+    Install Slackcat
+        ${result}=      Run Process  wget  https://github.com/vektorlab/slackcat/releases/download/v1.0/slackcat-1.0-linux-amd64  -O  slackcat
+        Sleep  5s
+        # Log To Console   \nSTDOUT: ${result.stdout} \nSTDERR: ${result.stderr} \nRC ${result.rc}
+        # Should Contain  ${result.stderr}  ‘slackcat’ saved
+        File Should Exist  slackcat 
+
+    Add slack token at ${ST2CHATOPS.ENV}
```

Comment:
Two things here: 

1. Moved the st2chatops.env modification to suite setup and teardown. The steps are:
     - Create another file _st2chatops.env.orig_ and then edit _st2chatops.env_.
     - After test suite execution is done move the _st2chatops.env.orig_ back as _st2chatops.env_.

2. The above steps take care of the append as well. I am not inclined to append any lines to the file, as other CI tests changes the `# export HUBOT_ADAPTER=slack` setting as `# export HUBOT_ADAPTER=shell`.